Sofa/framework/Core/src/sofa/core/objectmodel/BaseNode.h

```diff
@@ -54,6 +54,8 @@ class SOFA_CORE_API BaseNode : public virtual Base
     BaseNode() ;
     ~BaseNode() override;
 
+    BaseNode* m_root {nullptr} ;
```

Comment:
maybe you could use a ```unique_ptr<BaseNode>``` here?

Reply:
Aren't an unique_ptr a smart pointer that delete  the pointee when the unique_ptr is out of scope ? So this means that deleting a child node would led to the deletation the root node. 
If I'm not making a mistake I don't see any relevant use for an unique_ptr here. 

But maybe you are more thinking to a sofa::sptr ?

Reply:
@epernod ?

Reply:
done